snapcraft/plugins/_python/_pip.py

```diff
@@ -372,20 +373,31 @@ def list(self, *, user=False):
         if user:
             command.append('--user')
 
-        output = self._run_output(command)
         packages = collections.OrderedDict()
         try:
+            output = self._run_output(command)
             json_output = json.loads(
                     output, object_pairs_hook=collections.OrderedDict)
+            for package in json_output:
+                if 'name' not in package:
+                    raise errors.PipListMissingFieldError('name', output)
+                if 'version' not in package:
+                    raise errors.PipListMissingFieldError('version', output)
+                packages[package['name']] = package['version']
         except json.decoder.JSONDecodeError as e:
             raise errors.PipListInvalidJsonError(output) from e
+        except subprocess.CalledProcessError:
+            command = ['list']
+            if user:
+                command.append('--user')
+            output = self._run_output(command)
+            # --format requires a newer pip, so fall back to legacy output
+            version_regex = re.compile('\((.+)\)')
+            for line in output.splitlines():
+                line = line.split()
+                m = version_regex.search(line[1])
+                packages[line[0]] = m.group(1)
```

Comment:
can you recontruct `json_output` here instead and leave the logic below untouched?